pkg/vm/builtins/range_check.go

```diff
@@ -46,17 +48,18 @@ func (r *RangeCheck) GetAllocatedSize(segmentUsedSize uint64, vmCurrentStep uint
 	return getBuiltinAllocatedSize(segmentUsedSize, vmCurrentStep, r.ratio, inputCellsPerRangeCheck, instancesPerComponentRangeCheck, cellsPerRangeCheck)
 }
 
-func (r *RangeCheck) GetRangeCheckUsage(rangeCheckSegment *memory.Segment) (uint64, uint64) {
-	minVal, maxVal := ^uint64(0), uint64(0)
+// GetRangeCheckUsage returns the min and max values used in the range check segment. Since each range check instance consists of 16-bit parts, the min and max values are calculated by iterating over the segment data and extracting the 16-bit parts from each field element.
```

Comment:
// getPermRangeCheckLimits returns the minimum and maximum values used by the range check units in the program.

// GetRangeCheckUsage returns the min and max values used in the range check segment.

doesnt these sound the same?

Reply:
Nope, bc GetRangeCheckUsage checks the segment, and the getPermRangeCheckLimits checks "all" the range checks done during the execution, which includes range checks done on the instructions

Reply:
And range check done on instructions are not in the range check segment ?

Reply:
from what I understand they are not, because for instruction we are checking only if the offsets are in range 2**16, and add that to the trace, so then the virtual column with permutation range checks is constructed by the prover.

Reply:
There is a difference between regular range check and permutation range check, first one checks if the number is in desired range (for instance, normally we use 8 * 2**16 checks), where permutation range check is a single range check unit